core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -219,25 +242,83 @@ private void resetInputAndInternalAndSeekToEndIntermediateTopicOffsets() {
                 }
             }
 
-            if (inputAndInternalTopicPartitions.size() > 0) {
-                client.seekToBeginning(inputAndInternalTopicPartitions);
+            maybeSeekToBeginning(client, inputAndInternalTopicPartitions, internalTopics);
+
+            maybeSeekToEnd(client, intermediateTopicPartitions);
+
+            if (!dryRun) {
```

Comment:
@dguy as to have a single code path for main logic and dry-run logic it is done this way, 
The intial PR is done like a single check 
if dry-run 
{
}
else {
}

for more info can have a look into @mjsax comments. As the logic between dry-run and actual logic may change if any future changes will be done to this tool, which may show different behavior. So felt it is valid and followed this approach, even if the code does not look cleaner.